src/status_im/events.cljs

```diff
@@ -308,24 +310,31 @@
    (accounts.recover/validate-password cofx)))
 
 (handlers/register-handler-fx
- :accounts.recover.ui/sign-in-button-pressed
+ :accounts.recover.ui/recover-account-confirmed
  [(re-frame/inject-cofx :random-guid-generator)]
  (fn [cofx _]
-   (accounts.recover/recover-account-with-checks cofx)))
+   (accounts.recover/recover-account cofx)))
 
 (handlers/register-handler-fx
- :accounts.recover.ui/recover-account-confirmed
+ :accounts.recover.ui/recover-account-enter-password
  [(re-frame/inject-cofx :random-guid-generator)]
- (fn [cofx _]
-   (accounts.recover/recover-account cofx)))
+ (fn [cofx [_ masked-passphrase accounts]]
+   (accounts.recover/navigate-to-recover-account-password-screen (assoc cofx :masked-passphrase masked-passphrase :accounts accounts))))
 
 (handlers/register-handler-fx
  :accounts.recover.callback/recover-account-success
  [(re-frame/inject-cofx :random-guid-generator)
   (re-frame/inject-cofx :accounts.create/get-signing-phrase)
   (re-frame/inject-cofx :accounts.create/get-status)]
- (fn [cofx [_ result password]]
-   (accounts.recover/on-account-recovered cofx result password)))
+ (fn [cofx [_ result masked-passphrase password address name photo-path accounts error]]
+   (if (nil? password)
```

Comment:
Two nested `if`'s can be replaced with one `cond` for better readability.

Reply:
But that would mean repeating the (nil? password) test in both conditions, not sure I agree it's clearer? Could you provide an example please?